### ISSUE TITLE
Neural Networks Classifier

### ISSUE THREAD

**Issue by iLOSTA** [iLOSTA/The-Project#8]
I started looking at various ways to further enhance the already very high performance of this model. Among the methods I tried:

- Adding more layers 
- trying different feature extraction strategies.

I will detail each method in a separate comment.

**Comment by iLOSTA** [comment]
**ADDING MORE LAYERS**

The simple version of the network had only 1 hidden layer of 20 neurons and resulted in 99.2% validation acc. Training time 18 sec. 
I added two more hidden layers and changed the number of the neurons for the first layer. (layer1: 80 nodes, layer2: 60 nodes, layer3: 40 nodes). Validation acc jumped to 99.67%, of course at a cost. Training time increased to 50 sec.

**Comment by iLOSTA** [comment]
**FEATURE EXTRACTION IDEAS**

Instead of working with words, whether it was 1 or 2 grams or a combination of both, I tried working on the character level. TfidfVectorizer provides a very nice trick to specify an analyzer which tells the vectorizer what makes a feature in the text. Possible analyzers are 'word', 'char' and 'char_wb'. 

While 'word' and 'char' are clear on what they do, 'char_wb' looks at ngrams of characters within one word. For example, for 'toyota' a (2, 2) char_wb results in (' t', 'to', 'oy', yo', ot, 'ta', 'a ') as features. This is very interesting because now the model can detect words similarities. 

I tried (3, 4) char_wb and the result is very good. Although the increase in the validation accuracy is not big (99.67 -> 99.77), It still means something and is worth noting.

**Comment by iLOSTA** [comment]
The result of combining the above two steps resulted in a 0.05 increase of the model's accuracy.
